deepchecks/vision/checks/performance/performance_report.py

```diff
@@ -45,44 +45,56 @@ def __init__(self,
         super().__init__()
         self.alternative_metrics = alternative_metrics
         self.prediction_formatter = prediction_formatter
+        self._state = {}
 
-    def run_logic(self, context: Context) -> CheckResult:
-        """Run check.
-
-        Returns
-        -------
-        CheckResult
-            value is dictionary in format 'score-name': score-value
-        """
-        train_dataset = context.train
-        test_dataset = context.test
-        model = context.model
+    def initialize_run(self, context: Context):
+        """Optional method to initialize run before starting updating on batches."""
+        self._state = {}
         context.assert_task_type(TaskType.CLASSIFICATION, TaskType.OBJECT_DETECTION)
-
         # Get default scorers if no alternative, or validate alternatives
-        scorers = get_scorers_list(test_dataset, self.alternative_metrics)
-        datasets = {'Train': train_dataset, 'Test': test_dataset}
-
-        classes = train_dataset.get_samples_per_class().keys()
-        plot_x_axis = 'Class'
+        self._state['train'] = {}
+        self._state['test'] = {}
+        self._state['train']['scorers'] = get_scorers_list(context.train, self.alternative_metrics)
```

Comment:
why do you need to save all this properties (scorers, samples, classes) in the state?
They will be available in the context and they are cached anyway.